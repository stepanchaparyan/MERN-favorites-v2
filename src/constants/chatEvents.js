export const CHAT_EVENTS = {
  JOIN: 'join',
  MESSAGE_RECEIVED: 'messageReceived',
  USER_JOINED: 'userJoined',
  LEAVE_CHAT: 'leaveChat',
  SEND_MESSAGE: 'sendMessage'
};
