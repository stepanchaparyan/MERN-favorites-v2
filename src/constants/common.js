export const COMMON = {
  TEXT: {
    FILE_UPLOADED: 'File Uploaded',
  },
};
