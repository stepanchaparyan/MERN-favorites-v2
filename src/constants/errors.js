export const ERRORS = {
  TEXT: {
    PASSWORD_MATCH: 'Password does not match'
  },
  TEXT_500: 'There was a problem with the server'
};
