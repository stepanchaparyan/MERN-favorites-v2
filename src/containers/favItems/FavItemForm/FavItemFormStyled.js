import styled from 'styled-components';
import { tabletUp, desktopUp, desktopLargeUp } from '../../../styles/mediaQueries/mixins';
import { Form, Field } from 'formik';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  color: ${props => props.theme.cadetblue};
  z-index: 1;
  position: absolute;
  background-color: gray;
  left: 50%;
  top: 24%;
  transform: translate(-50%, -50%);
  border-radius: 4px;
  min-width: 66%;
  min-height: 20%;
  text-align: center;
  font-size: 14px;
  ${tabletUp`
    top: 36%;
    min-width: 50%;
    min-height: 30%;
  `};
  ${desktopUp`
    top: 36%;
    min-width: 40%;
    min-height: 32%;
  `};
  ${desktopLargeUp`
    top: 50%;
    min-width: 36%;
    min-height: 48%;
  `};
`;

export const Title = styled.div`
  display: flex;
  justify-content: center;
  padding: 32px;
  font-weight: 600;
  font-size: 22px;
  color: black;
  ${tabletUp`
    font-size: 24px;
  `};
  ${desktopUp`
    font-size: 26px;
  `};
  ${desktopLargeUp`
    font-size: 30px;
  `};
`;

export const Input = styled.input`
  border: 0px;
  width: 200px;
  padding: 8px 10px;
  margin: 4px;
  outline: none;
  border-radius: 4px;
`;

export const ButtonSubmit = styled.button`
  padding: 8px 10px;
  margin: 4px;
  outline: none;
  border-radius: 4px;
  width: 20%;
  border: 0;
  background-color: ${props => props.theme.cadetblue};
  font-weight: 600;
  font-size: 16px;
  margin-top: 20px;
  cursor: pointer;
  margin-bottom: 10px;
`;

export const CancelButton = styled.button`
  padding: 8px 10px;
  margin: 4px;
  outline: none;
  border-radius: 4px;
  width: 30%;
  border: 0;
  background-color: gray;
  font-weight: 600;
  font-size: 16px;
  cursor: pointer;
  margin-bottom: 10px;
`;

export const Errors = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 10px 10px 10px;
`;

export const ErrorButton = styled.button`
  color: red;
  background-color: transparent;
  border: none;
  outline: none;
  cursor: pointer;
`;

export const StyledSelectField = styled(Field)`
  width: 58%;
  background: white;
  height: 32px;
  outline: none;
  border: none;
  border-bottom: 1px solid cadetblue;
  padding-left: 5px;
  margin: 4px;
  ${tabletUp`
    width: 45%;
  `};
`;

export const Option = styled.option`
  color: black;
  height: 47px;
  padding: 20px;
  font-size: 18px;
  color: ${props => props.theme.cadetblue};
`;

export const DefaultOption = styled.option`
  display: none;
`;

export const FormStyled = styled(Form)`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const FieldStyled = styled(Field)`
  border: 0;
  border-bottom: 1px solid ${props => props.theme.cadetblue};
  width: 50%;
  padding: 7px 10px;
  margin: 3px;
  outline: none;
  ${tabletUp`
    width: 40%;
  `};
`;

export const ErrorMessages = styled.div`
  color: red;
  font-size: 12px;
`;
