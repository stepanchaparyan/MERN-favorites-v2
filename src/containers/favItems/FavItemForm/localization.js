import { defineMessages } from 'react-intl';

export default defineMessages({
  selectCategory: {
    id: 'favItem.selectCategory',
    defaultMessage: 'Select Category',
  },
  films: {
    id: 'favItem.films',
    defaultMessage: 'Films',
  },
  music: {
    id: 'favItem.music',
    defaultMessage: 'Music',
  },
  books: {
    id: 'favItem.books',
    defaultMessage: 'Books',
  },
  other: {
    id: 'favItem.other',
    defaultMessage: 'Other',
  },
  removeCard: {
    id: 'favItem.removeCard',
    defaultMessage: 'Remove Card',
  },
  editFavItem: {
    id: 'favItem.editFavitem',
    defaultMessage: 'Edit favorite item',
  },
  addNewFavItem: {
    id: 'favItem.addNewFavItem',
    defaultMessage: 'Add new favorite item',
  },
  author: {
    id: 'favItem.author',
    defaultMessage: 'Author',
  },
  title: {
    id: 'favItem.title',
    defaultMessage: 'Title',
  },
  description: {
    id: 'favItem.description',
    defaultMessage: 'Description',
  },
  update: {
    id: 'favItem.update',
    defaultMessage: 'Update',
  },
  add: {
    id: 'favItem.add',
    defaultMessage: 'Add',
  },
  cancel: {
    id: 'favItem.cancel',
    defaultMessage: 'Cancel',
  },
});
