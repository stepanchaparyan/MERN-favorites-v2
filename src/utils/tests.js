import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';

export function renderToJSON(Component, props = {}, Router) {
  if (Router) {
    return renderer
      .create(
        <Router>
          <Component {...props} />
        </Router>
      )
      .toJSON();
  }
  return renderer.create(<Component {...props} />).toJSON();
}

export function renderSnapshot(Component, props = {}) {
  const wrapper = shallow(<Component {...props} />);
  expect(wrapper).toMatchSnapshot();
}
