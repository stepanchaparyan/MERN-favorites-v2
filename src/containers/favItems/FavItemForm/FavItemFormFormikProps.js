import * as Yup from 'yup';
import { FAVITEM } from '../../../constants';

const { DEFAULT_VALUES } = FAVITEM;

const defaultValues = {
  author: DEFAULT_VALUES.AUTHOR,
  title: DEFAULT_VALUES.TITLE,
  category: DEFAULT_VALUES.CATEGORY,
  description: DEFAULT_VALUES.DESCRIPTION,
};

const validationSchema = Yup.object({
  author: Yup.string().required('Required'),
  title: Yup.string()
    .min(2, 'Title must have min 2 symbols')
    .required('Required'),
  category: Yup.string().required('Required'),
  description: Yup.string().required('Required'),
});

const formikProps = editFavItem => ({
  initialValues: editFavItem ? editFavItem : defaultValues,
  validateOnChange: true,
  validationSchema,
});

export default formikProps;
