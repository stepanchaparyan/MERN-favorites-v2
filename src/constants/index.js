import { FAVITEM } from './favItem';
import { FORM } from './form';
import { EVENTS } from './events';
import { REACT_TOOLTIP_STYLED } from './reactTooltipStyled';
import { HEADER_CONFIG } from './headers';
import { LINK } from './link';
import { LANGUAGES } from './languages';
import { MODAL } from './modal';
import { FILE_UPLOAD } from './fileUpload';
import { URL } from './url';
import { ERRORS } from './errors';
import { COMMON } from './common';

const NOOP = () => {};

export {
  FAVITEM,
  FORM,
  EVENTS,
  REACT_TOOLTIP_STYLED,
  HEADER_CONFIG,
  LINK,
  LANGUAGES,
  MODAL,
  FILE_UPLOAD,
  URL,
  ERRORS,
  COMMON,
  NOOP
};
