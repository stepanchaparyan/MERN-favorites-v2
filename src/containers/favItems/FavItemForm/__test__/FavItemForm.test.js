import React, { useContext } from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { renderSnapshot } from '../../../../utils/tests';
import FavItemForm from '../FavItemForm.js';
import { FieldStyled, ErrorButton } from '../FavItemFormStyled';
import { FAVITEM } from '../../../../constants';

const { DEFAULT_VALUES } = FAVITEM;

jest.mock('react-intl', () => ({
  useIntl: jest.fn(() => ({ formatMessage: m => m.defaultMessage })),
  defineMessages: i => i
}));

jest.mock('react', () => {
  const ActualReact = jest.requireActual('react');
  return {
    ...ActualReact,
    useContext: jest.fn().mockImplementation(() => ({
      onSubmit: jest.fn(),
      update_FavItem: jest.fn(),
      error: 'a',
      clearErrors: jest.fn()
    }))
  };
});

describe('FavItemForm component test with Enzyme', () => {
  test('renders correct snapshot', () => {
    renderSnapshot(FavItemForm);
  });

  test('render submit on Form component', () => {
    const component = shallow(<FavItemForm />);
    const form = component.find('Formik');
    form.simulate('submit', { target: { name: '123' } });
  });

  test('should pass the correct initial values', () => {
    const component = shallow(<FavItemForm />);
    const initialValues = component.find('Formik').props().initialValues;

    expect(initialValues.author).toBe(DEFAULT_VALUES.AUTHOR);
    expect(initialValues.title).toBe(DEFAULT_VALUES.TITLE);
    expect(initialValues.category).toBe(DEFAULT_VALUES.CATEGORY);
    expect(initialValues.description).toBe(DEFAULT_VALUES.DESCRIPTION);
  });

  test('should pass the correct initial values when editFavitem is true', () => {
    useContext.mockImplementationOnce(
      jest.fn(() => ({
        editFavItem: { author: 'Joe' }
      }))
    );

    const component = shallow(<FavItemForm />);
    const initialValues = component.find('Formik').props().initialValues;

    expect(initialValues.author).toBe('Joe');
  });

  test('render change on Input component', () => {
    useContext.mockImplementationOnce(
      jest.fn(() => ({
        editFavItem: {},
        clearEdit: jest.fn(),
        toggle_Form: jest.fn(),
        update_FavItem: jest.fn(),
        clearErrors: jest.fn()
      }))
    );

    const component = shallow(<FavItemForm />);

    const inputs = component.find(FieldStyled);
    inputs.forEach(input => {
      input.simulate('change', { target: { value: 'target' } });
    });

    expect(component).toMatchSnapshot();
  });

  test('render click on ErrorButton when error is not array', () => {
    const component = shallow(<FavItemForm />);

    const errorButtons = component.find(ErrorButton);
    errorButtons.forEach(errorButton => {
      errorButton.simulate('click');
    });
    expect(component).toMatchSnapshot();
  });

  test('render click on ErrorButton when error is array', () => {
    useContext.mockImplementationOnce(
      jest.fn(() => ({
        editFavItem: {},
        clearEdit: jest.fn(),
        toggle_Form: jest.fn(),
        update_FavItem: jest.fn(),
        clearErrors: jest.fn(),
        error: ['a', 'b']
      }))
    );

    const component = shallow(<FavItemForm />);

    const errorButtons = component.find(ErrorButton);
    errorButtons.forEach(errorButton => {
      errorButton.simulate('click');
    });
    expect(component).toMatchSnapshot();
  });
});
