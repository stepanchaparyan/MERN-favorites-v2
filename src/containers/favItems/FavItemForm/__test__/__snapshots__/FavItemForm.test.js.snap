// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavItemForm component test with Enzyme render change on Input component 1`] = `
<styled.div>
  <styled.div>
    Edit favorite item
  </styled.div>
  <Formik
    initialValues={Object {}}
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_excludedEdges": Array [],
        "_mutate": undefined,
        "_nodes": Array [
          "description",
          "category",
          "title",
          "author",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "author": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "category": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "description": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "title": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "min": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <Component />
  </Formik>
</styled.div>
`;

exports[`FavItemForm component test with Enzyme render click on ErrorButton when error is array 1`] = `
<styled.div>
  <styled.div>
    Edit favorite item
  </styled.div>
  <Formik
    initialValues={Object {}}
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_excludedEdges": Array [],
        "_mutate": undefined,
        "_nodes": Array [
          "description",
          "category",
          "title",
          "author",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "author": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "category": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "description": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "title": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "min": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <Component />
  </Formik>
</styled.div>
`;

exports[`FavItemForm component test with Enzyme render click on ErrorButton when error is not array 1`] = `
<styled.div>
  <styled.div>
    Edit favorite item
  </styled.div>
  <Formik
    initialValues={
      Object {
        "author": "Unknown",
        "category": "Other",
        "description": "Description",
        "title": "",
      }
    }
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_excludedEdges": Array [],
        "_mutate": undefined,
        "_nodes": Array [
          "description",
          "category",
          "title",
          "author",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "author": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "category": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "description": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "title": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "min": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <Component />
  </Formik>
</styled.div>
`;

exports[`FavItemForm component test with Enzyme renders correct snapshot 1`] = `
<styled.div>
  <styled.div>
    Edit favorite item
  </styled.div>
  <Formik
    initialValues={
      Object {
        "author": "Unknown",
        "category": "Other",
        "description": "Description",
        "title": "",
      }
    }
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_excludedEdges": Array [],
        "_mutate": undefined,
        "_nodes": Array [
          "description",
          "category",
          "title",
          "author",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "author": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "category": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "description": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "title": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "min": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <Component />
  </Formik>
</styled.div>
`;
